### ISSUE TITLE
Skripta za treniranje i dataset

### ISSUE THREAD

**Issue by MarkoStaletovic** [Drakula44/LPCV_2025_T1#21]
Napravio sam na branchu MarkoStaletovic/DatasetCroppingAndUtils klasu koja na osnovu datog dataseta (za koji sam za sada samo koristio validacioni) i anotacije dobijam isecke slika koji sadrze samo zeljeni objekat. Kada pokusam da iskoristim skriptu za treniranje ne dobijam da loss opada kroz epohe.

Pokušano:

1. Promena batch size-a i learning rate-a
2. Testiranje i na GPU-u i na CPU-u

Iskreno nemam pojma sta je uzrok, ali moze biti nesto od sledeceg:

1. Cropovane slike su vrlo cesto loseg kvaliteta, jer su samo mali deo slike koji je resizeovan i razvucen
2. Postoji neki problem kod normalizacije na pocetku (iskreno ne verujem da je ovo, trebalo bi transform uvek radi isto)
3. Trenutno skripta za treniranje trenira sve layere (to jest fine-tunuje i sve one pre poslednjeg klasifikatora), sto mozda u kombinaciji sa losim + malim datasetom pravi da se model dosta gubi, jer su imagenet slike 224x224 ali izgledaju normalno i nisu deformisane
4. Postoji neki bug u loaderu za dataset ili u skripti za treniranje
